### DISCUSSION TITLE
Doesn't check https on port 443

### DISCUSSION THREAD

**Discussion by zPrototype** [projectdiscovery/httpx#293] [Q&A] [answered]
When running httpx with a list of (sub)domains and specifying port 443 it probes http which results in a 400 Bad Request. Shouldn't it automatically use https here or at least try it?
![httpx_defunct](https://user-images.githubusercontent.com/43825360/120053605-5c52f780-c02b-11eb-88af-3dcf8bc05e73.png)

Command used is:
"""
httpx -l domains -silent -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:55.0) Gecko/20100101 Firefox/55.0' -ports 80,8080,8081,8443,443,7001,3000 -status-code -no-color -follow-redirects -title -websocket -o httpx_subs2.txt
"""

**Comment by ehsandeep** [comment, marked as answer]
@zPrototype httpx, as default probes for **https** and if failed, it attempts to **http**, so if you see results on http, I assume https was not accessible for that target, otherwise do you have an example to share where **https** is accessible but you still see http results?

**Reply by zPrototype** [reply]
Did some more testing and you are right! Everything works as intended. This can be marked as closed :)
